src/web/app/components/sortable-table/sortable-table.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { SortOrder } from '../../../types/sort-properties';
+
+/**
+ * Column data for sortable table
+ */
+export interface ColumnData<T> {
+  header: string;
+  headerToolTip?: string;
+  sortBy?(item1: T, item2: T): number; // Don't provide this field if you don't want this column to be sortable
+}
+
+/**
+ * Displays a sortable table, sorting by clicking on the header
+ * Optional sorting function to be inputted provided for each column
+ * Columns and rows provided must be aligned
+ */
+@Component({
+  selector: 'tm-sortable-table',
+  templateUrl: './sortable-table.component.html',
+  styleUrls: ['./sortable-table.component.scss'],
+})
+export class SortableTableComponent implements OnInit {
+
+  @Input()
+  columns: ColumnData<any>[] = [];
+
+  @Input()
+  rows: any[][] = [];
```

Comment:
Not sure if we can use typescript to enforce that the types of items in the columns and rows arrays are properly aligned. Searched around for quite a bit and couldn't find any solutions.